src/shogun/base/SGObject.cpp

```diff
@@ -1119,4 +1119,12 @@ std::string CSGObject::string_enum_reverse_lookup(
 		    return p.second == enum_value;
 	    });
 	return enum_map_it->first;
-}
\ No newline at end of file
+}
+
+ObservedValue::ObservedValue(int64_t step, std::string name)
+    : CSGObject(), m_step(step), m_name(name), m_any_value(Any())
+{
+	SG_ADD(&m_step, "step", "Step");
+	this->watch_param(
```

Comment:
do we need to store this? Is this ever used with a put/get, by user or developer? If no, remove

Reply:
The name is needed because the observers use it to filter the `ObservedValue` they receive such to keep only the ones they are looking for (e.g I'm looking for "w", therefore I will process only those `ObservedValue` which have it as a name ).